test/unit/plugin_test.rb

```diff
@@ -424,6 +424,25 @@ def test_hosts_controller_action_scope_added_to_local
     assert_equal 2, scopes.count
   end
 
+  def test_add_resource_permissions_to_core
+    Foreman::Plugin.register :test_plugin do
+      add_resource_permissions_to_core ["Test::Resource"], :except => [:create_test]
+    end
+    manager = Role.find_by :name => "Manager"
+    viewer = Role.find_by :name => "Viewer"
+    assert_equal 2, manager.permissions.where(:resource_type => "Test::Resource").count
```

Comment:
Perhaps related to comment above, 3 out of 4 fixtures I added are associated with the resource (view_test, create_test, edit_test). `:except => [:create_test]` says I do not want to add create_test permission.